sdk/cosmos/azure_data_cosmos/src/retry_policies/mod.rs

```diff
@@ -66,3 +71,24 @@ pub trait RetryPolicy: Send + Sync {
     /// A `RetryResult` indicating the retry decision.
     async fn should_retry(&mut self, response: &azure_core::Result<RawResponse>) -> RetryResult;
 }
+
+fn get_substatus_code_from_error(err: &azure_core::Error) -> SubStatusCode {
+    if let ErrorKind::HttpResponse { raw_response, .. } = err.kind() {
+        raw_response
+            .as_ref()
+            .and_then(|r| r.headers().get_as(&SUB_STATUS).ok())
+            .and_then(|raw: u16| SubStatusCode::try_from(raw).ok())
```

Comment:
If you implement the `FromHeaders` trait (in `azure_core::http`) on `SubStatusCode`, you'll be able to replace this with a single call to `r.headers().get::<SubStatusCode>()`. The `FromHeaders` trait allows the `SubStatusCode` to specify the header name to read AND how to read it, so it all becomes self-describing and strongly-typed :).